packages/insomnia-sdk/src/objects/folders.ts

```
@@ -0,0 +1,48 @@
+import { Environment } from './environments';
+
+// Folder reprensents a request folder in Insomnia.
+export class Folder {
+    id: string;
+    name: string;
+    environment: Environment;
+
+    constructor(id: string, name: string, environmentObject: object | undefined) {
+        this.id = id;
+        this.name = name;
+        this.environment = new Environment(`${id}.environment`, environmentObject);
+    }
+
+    toObject = () => {
+        return {
+            id: this.id,
+            name: this.name,
+            environment: this.environment.toObject(),
+        };
+    };
+}
+
+// ParentFolders reprensents ancestor folders of the active request
+export class ParentFolders {
+    constructor(private folders: Folder[]) { }
+
+    get = (idOrName: string) => {
+        return this.folders.find(folder => folder.name === idOrName || folder.id === idOrName);
+    };
+
+    getById = (id: string) => {
+        return this.folders.find(folder => folder.id === id);
+    };
+
+    getByName = (folderName: string) => {
+        return this.folders.find(folder => folder.name === folderName);
```

It is better to throw an exception telling use the folder is not found when the name is invalid.
Otherwise, when using ``insomnia.parentFolders.getByName(XXX).environment`` with an invalid name, it will show the following message which is not very clear.
```
message: Cannot read properties of undefined (reading 'environment'); stack: TypeError: Cannot read properties of undefined (reading 'environment')
at fn0 (eval at runScript (http://localhost:3334/hidden-window.ts:43:25), <anonymous>:7:62)
at eval (eval at runScript (http://localhost:3334/hidden-window.ts:43:25), <anonymous>:9:13)
at runScript (http://localhost:3334/hidden-window.ts:61:39)
at async http://localhost:3334/hidden-window.ts:19:20
```

---

Good points, updated, snippets are originally scoped as another task and they are also added in this PR.